docs/iam-user-service.md

```diff
@@ -0,0 +1,119 @@
+- The name with which the parameter is created `/iam_users/<user-name>_keys`.
+- The keys are stored in the parameter store in the below json structure
+
+```json
+{
+  "pending_deletion": {
+    "AccessKeyID": "Access_Key_ID_Pending_Deletion",
+    "SecretAccessKey": "Secret_Access_Key_Pending_Deletion"
+  },
+  "current": {
+    "AccessKeyID": "Access_Key_ID_Current",
+    "SecretAccessKey": "Secret_Access_Key_Current"
+  }
+}
+```
+
+![parameter](/images/iam-user-service/parameter.png)
+
+## Setup automation to retrieve and use keys
+
+Users can implement a checker in their automation scripts to verify if they are using a 'pending_deletion' key. If so, the script should automatically rotate to the 'current' key fetched from the Parameter Store.
+
+This way one can stay ahead of the key rotation and always use the current keys.
+
+## Permission boundaries
+
+Permission boundaries prevent privilege escalation by setting the maximum permissions users can have, even if their IAM policies allow more extensive permissions. For instance, an `AdministratorAccess` policy constrained by a permission boundary that only allows `AmazonS3FullAccess` will limit the user to S3 actions.
+
+The lambda function automatically set's permission boundaries to the user's created as a security measure.
+
+### Deleting an IAM user
+
+Remove the corresponding entry from the DynamoDB table. The Lambda function will trigger and delete the user and their access keys from IAM and the SSM Parameter Store.
+
+## Best practices
+
+### Applying least-privilege permissions
+
+When setting permissions with IAM policies, grant only the permissions needed for a task. Define actions on specific resources under specific conditions, known as least-privilege permissions.
+
+Initially, you might have broad permissions to explore what's needed for your workload. As your use case matures, work to reduce permissions and move towards least privilege.
+
+### Deny access to AWS based on the source IP
+
+In AWS we can create identity-based policy that denies access to AWS actions in the account when the request comes from principals outside the specified IP range.
+
+Example policy that allows s3 actions only to a specific ip range:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "*"
+        },
+        {
+            "Effect": "Deny",
+            "Action": "*",
+            "Resource": "*",
+            "Condition": {
+                "NotIpAddress": {
+                    "aws:SourceIp": [
+                        "192.0.2.0/24"
+                    ]
+                },
+                "Bool": {
+                    "aws:ViaAWSService": "false"
+                }
+            }
+        }
+    ]
+}
+```
+
+<!-- Related pages need to be added here  -->
```

Comment:
Please make sure to add any supporting documents or links that can help users obtain help related to the topic